packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.ct.tsx

```diff
@@ -2,65 +2,77 @@ import { expect, test } from "../../playwright-axe";
 import OnyxCheckbox from "./OnyxCheckbox.vue";
 
 test("should render unchecked", async ({ mount, makeAxeBuilder }) => {
-  // ARRANGE
-  const component = await mount(
-    <div style="display: grid; width: max-content;">
-      <OnyxCheckbox label="Unchecked" />
-      <OnyxCheckbox label="Hover" />
-      <OnyxCheckbox label="Focus visible" />
-    </div>,
-  );
+  const testCases = [{ name: "default" }, { name: "required", required: true }];
 
-  // ACT
-  const accessibilityScanResults = await makeAxeBuilder().analyze();
+  for (const testCase of testCases) {
+    // ARRANGE
+    const component = await mount(
+      <div style="display: grid; width: max-content;">
+        <OnyxCheckbox label={`Unchecked ${testCase.name}`} required={testCase.required} />
+        <OnyxCheckbox label="Hover" required={testCase.required} />
+        <OnyxCheckbox label="Focus visible" required={testCase.required} />
+      </div>,
+    );
 
-  // ASSERT
-  expect(accessibilityScanResults.violations).toEqual([]);
+    // ACT
+    const accessibilityScanResults = await makeAxeBuilder().analyze();
 
-  const checkboxes = await component.getByRole("checkbox").all();
+    // ASSERT
+    expect(accessibilityScanResults.violations).toEqual([]);
 
-  // ACT
-  await checkboxes[1].hover();
-  await checkboxes[2].focus();
+    const checkboxes = await component.getByRole("checkbox").all();
 
-  // ASSERT
-  for (const checkbox of checkboxes) {
-    await expect(checkbox).not.toBeChecked();
-  }
+    // ACT
+    await checkboxes[1].hover();
+    await checkboxes[2].focus();
```

Comment:
wouldn't it make more sense to swap those around? so we would get a warning if our hover/focus styles break accessibility?

```suggestion
    const checkboxes = await component.getByRole("checkbox").all();
    await checkboxes[1].hover();
    await checkboxes[2].focus();
    
    // ACT
    const accessibilityScanResults = await makeAxeBuilder().analyze();    

    // ASSERT
    expect(accessibilityScanResults.violations).toEqual([]);
```

Reply:
I had flaky firefox tests with this order so I had to do the accessibility test before

Reply:
whoah.. then let's just hope that the hover and focus still have enough contrast etc ^^